include/ada/url-inl.h

```diff
@@ -96,39 +100,39 @@ inline std::ostream &operator<<(std::ostream &out, const ada::url &u) {
     running_index += helpers::fast_digit_count(*port) + 1;  // Port omits ':'
   }
 
-  out.pathname_start = uint32_t(running_index);
+  out.pathname_start = static_cast<uint32_t>(running_index);
 
   running_index += path.size();
 
   if (query.has_value()) {
-    out.search_start = uint32_t(running_index);
+    out.search_start = static_cast<uint32_t>(running_index);
     running_index += get_search().size();
-    if (get_search().size() == 0) {
+    if (get_search().empty()) {
       running_index++;
     }
   }
 
   if (hash.has_value()) {
-    out.hash_start = uint32_t(running_index);
+    out.hash_start = static_cast<uint32_t>(running_index);
   }
 
   return out;
 }
 
 inline void url::update_base_hostname(std::string_view input) { host = input; }
 
-inline void url::update_unencoded_base_hash(std::string_view input) {
+inline void url::update_unencoded_base_hash(std::string_view const input) {
   // We do the percent encoding
   hash = unicode::percent_encode(input,
                                  ada::character_sets::FRAGMENT_PERCENT_ENCODE);
 }
 
-inline void url::update_base_search(std::string_view input,
+inline void url::update_base_search(std::string_view const input,
                                     const uint8_t query_percent_encode_set[]) {
   query = ada::unicode::percent_encode(input, query_percent_encode_set);
 }
 
-inline void url::update_base_search(std::optional<std::string> input) {
+inline void url::update_base_search(std::optional<std::string> const &input) {
```

Comment:
That's potentially an interesting change. By passing the std::optional by value, it is likely that we create a whole new copy of the string.

Good!!!!

Reply:
In the library, we put the const qualifier first... so it should be...

```
inline void url::update_base_search(const  std::optional<std::string> &input) {
```

Reply:
You could use clang-format for that and change everything in one go.

Here's the option that you need for clang-format in order to reorder the qualifiers:
![image](https://github.com/ada-url/ada/assets/12122474/757ddc7b-f0b4-44b9-aff2-508ad41a4904)